compiler/kernel/op.go

```diff
@@ -42,14 +43,24 @@ var ErrJoinParents = errors.New("join requires two upstream parallel query paths
 type Builder struct {
 	pctx     *op.Context
 	source   *data.Source
-	planners map[dag.Source]from.Planner
+	listers  map[dag.Source]*meta.Lister
+	pools    map[dag.Source]*lake.Pool
+	progress *zbuf.Progress
+	deletes  *sync.Map
 }
 
 func NewBuilder(pctx *op.Context, source *data.Source) *Builder {
 	return &Builder{
-		pctx:     pctx,
-		source:   source,
-		planners: make(map[dag.Source]from.Planner),
+		pctx:    pctx,
+		source:  source,
+		listers: make(map[dag.Source]*meta.Lister),
+		pools:   make(map[dag.Source]*lake.Pool),
+		progress: &zbuf.Progress{
+			BytesRead:      0,
+			BytesMatched:   0,
+			RecordsRead:    0,
+			RecordsMatched: 0,
+		},
```

Comment:
Nit: No need to initialize fields to the zero value.
```suggestion
		progress: &zbuf.Progress{},
```